pkg/agent/agent_test.go

```diff
@@ -134,3 +141,155 @@ func TestGetRoundInfo(t *testing.T) {
+		},
+		{
+			name:             "hybrid mode",
+			trafficEncapMode: config.TrafficEncapModeHybrid,
+			expectedNodeConfig: config.NodeConfig{
+				Name:            nodeName,
+				OVSBridge:       ovsBridge,
+				DefaultTunName:  defaultTunInterfaceName,
+				PodIPv4CIDR:     podCIDR,
+				NodeIPAddr:      nodeIPNet,
+				NodeMTU:         1500,
+				UplinkNetConfig: new(config.AdapterNetConfig),
+			},
+			expectedNodeAnnotation: map[string]string{types.NodeMACAddressAnnotationKey: macAddr.String()},
+		},
+		{
+			name:             "encap mode",
+			trafficEncapMode: config.TrafficEncapModeEncap,
+			expectedNodeConfig: config.NodeConfig{
+				Name:            nodeName,
+				OVSBridge:       ovsBridge,
+				DefaultTunName:  defaultTunInterfaceName,
+				PodIPv4CIDR:     podCIDR,
+				NodeIPAddr:      nodeIPNet,
+				NodeMTU:         1500,
+				UplinkNetConfig: new(config.AdapterNetConfig),
+			},
+			expectedNodeAnnotation: nil,
+		},
+		{
+			name:             "encap mode, geneve tunnel",
+			trafficEncapMode: config.TrafficEncapModeEncap,
+			tunnelType:       ovsconfig.GeneveTunnel,
+			expectedNodeConfig: config.NodeConfig{
+				Name:            nodeName,
+				OVSBridge:       ovsBridge,
+				DefaultTunName:  defaultTunInterfaceName,
+				PodIPv4CIDR:     podCIDR,
+				NodeIPAddr:      nodeIPNet,
+				NodeMTU:         1450,
+				UplinkNetConfig: new(config.AdapterNetConfig),
+			},
+			expectedNodeAnnotation: nil,
+		},
+		{
+			name:             "encap mode, mtu specified",
+			trafficEncapMode: config.TrafficEncapModeEncap,
+			tunnelType:       ovsconfig.GeneveTunnel,
+			mtu:              1400,
+			expectedNodeConfig: config.NodeConfig{
+				Name:            nodeName,
+				OVSBridge:       ovsBridge,
+				DefaultTunName:  defaultTunInterfaceName,
+				PodIPv4CIDR:     podCIDR,
+				NodeIPAddr:      nodeIPNet,
+				NodeMTU:         1400,
+				UplinkNetConfig: new(config.AdapterNetConfig),
+			},
+			expectedNodeAnnotation: nil,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			node := &corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: nodeName,
+				},
+				Spec: corev1.NodeSpec{
+					PodCIDR: podCIDRStr,
+				},
+				Status: corev1.NodeStatus{
+					Addresses: []corev1.NodeAddress{
+						{
+							Type:    corev1.NodeInternalIP,
+							Address: nodeIPStr,
+						},
+					},
+				},
+			}
+			client := fake.NewSimpleClientset(node)
+			ifaceStore := interfacestore.NewInterfaceStore()
+			defer mockGetIPNetDeviceFromIP(nodeIPNet, ipDevice)()
+			defer mockNodeNameEnv(nodeName)()
+
+			initializer := &Initializer{
+				client:     client,
+				ifaceStore: ifaceStore,
+				mtu:        tt.mtu,
+				ovsBridge:  ovsBridge,
+				networkConfig: &config.NetworkConfig{
+					TrafficEncapMode: tt.trafficEncapMode,
+					TunnelType:       tt.tunnelType,
+				},
+			}
+			assert.NoError(t, initializer.initNodeLocalConfig())
```

Comment:
nit: this should probably be a `require`. If `initNodeLocalConfig` fails, `initializer.nodeConfig` could be `nil`?

Reply:
Right, thanks for catching it